kufar/books/main.go

```diff
@@ -209,12 +239,63 @@ func (d Database) SignInUser(w http.ResponseWriter, r *http.Request) {
 	cred.Username = credReq.Username
 
 	session.Values["user"] = cred.UUID
+
 	if err = session.Save(r, w); err != nil {
 		http.Error(w, http.StatusText(http.StatusInternalServerError)+": could not assign cookie", http.StatusInternalServerError)
 		return
 	}
+	w.Write([]byte(http.StatusText(http.StatusOK)))
+}
+
+//CheckAuth ...
+func (d Database) CheckAuth(w http.ResponseWriter, r *http.Request, userid int) bool {
+	cred := r.Header.Get("Authorization")
+	if cred == "" {
+		http.Error(w, http.StatusText(http.StatusInternalServerError)+": using cookies", http.StatusInternalServerError)
+		return false
+	}
+
+	s := strings.Split(cred, " ")
+
+	if len(s) != 2 || s[0] != "Basic" || s[1] == "" {
+		http.Error(w, http.StatusText(http.StatusUnauthorized)+": incorrect data. ", http.StatusUnauthorized)
+		return false
+	}
+
+	b, err := base64.StdEncoding.DecodeString(s[1])
+	bStr := string(b)
```

Comment:
Please change order of 266 and 267 lines.
You should at first check for err and then work with given `b` var